daemon/buildinfo.c

```diff
@@ -200,8 +200,67 @@
 
 #define FEAT_YES_NO(x) ((x) ? "YES" : "NO")
 
+// coverity[ +tainted_string_sanitize_content : arg-0 ]
+static inline void coverity_remove_taint(char *s)
+{
+    (void)s;
+}
+
+void read_install_type(void) {
+    char *filepath;
+    filepath = mallocz(sizeof(char) * (strlen(netdata_configured_user_config_dir) + strlen(".install-type") + 2));
+    sprintf(filepath, "%s/%s", netdata_configured_user_config_dir, ".install-type");
+    FILE *fp = fopen(filepath, "r");
+
+    if(fp) {
+        char *line = mallocz(sizeof(char) * (200 + 1));
+        while (fgets(line, 200, fp) != NULL) {
+            char *value = line;
+
+            while (*value && *value != '=') value++;
+
+            if (*value == '=') {
+                *value = '\0';
+                value++;
+                if (*value == '\'' || *value == '\"') value++;
+                char *end = value;
+                while (*end && *end != '\'' && *end != '\"' && *end != '\n') end++;
+                *end = '\0';
+                coverity_remove_taint(line);
+                coverity_remove_taint(value);
+            } else {
+                continue;
+            }
+
+            setenv(line, value, 1);
+        }
+        fclose(fp);
+        freez(line);
+    } else {
+        setenv("INSTALL_TYPE", "unknown", 1);
+    }
+
+    freez(filepath);
+}
+
 void print_build_info(void) {
+    read_install_type();
+    char *install_type = getenv("INSTALL_TYPE");
```

Comment:
I worry a bit about the use of environment var here... `INSTALL_TYPE` is quite generic are we not worried somebody else sets it thus making the `install_type == NULL` check not working as intended?

Reply:
Excellent point.

Were it almost anything but C, I would just be passing the values back to the caller, but doing that with multiple strings in C is not exactly easy.

Reply:
is there any other reason for doing it with env vars other than fighting with C? (I know you are not C dev... so maybe I can help you)

Reply:
Mostly fighting with C, but also because the code here is so far from being performance critical that the performance hit from manipulating the environment should never matter.